collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_host.py

```diff
@@ -627,49 +715,70 @@ def main():
     if module.params['nqn'] is not None and NVME_API_VERSION not in api_version:
         module.fail_json(msg='NVMe protocol not supported. Please upgrade your array.')
     state = module.params['state']
-    host = get_host(module, array)
-    if module.params['lun'] and not 1 <= module.params['lun'] <= 4095:
-        module.fail_json(msg='LUN ID of {0} is out of range (1 to 4095)'.format(module.params['lun']))
-    if module.params['volume']:
-        try:
-            array.get_volume(module.params['volume'])
-        except Exception:
-            module.fail_json(msg='Volume {0} not found'.format(module.params['volume']))
-    if module.params['preferred_array']:
-        try:
-            if module.params['preferred_array'] != ['delete']:
-                all_connected_arrays = array.list_array_connections()
-                if not all_connected_arrays:
-                    module.fail_json(msg='No target arrays connected to source array - preferred arrays not possible.')
-                else:
-                    current_arrays = [array.get()['array_name']]
-                    api_version = array._list_available_rest_versions()
-                    if NVME_API_VERSION in api_version:
-                        for current_array in range(0, len(all_connected_arrays)):
-                            if all_connected_arrays[current_array]['type'] == "sync-replication":
-                                current_arrays.append(all_connected_arrays[current_array]['array_name'])
+    if module.params['count']:
+        if not HAS_PURESTORAGE:
```

Comment:
This `if` statement should be moved outside of the `if module.params['count']` context.

Reply:
@dnix101 Why? We only care about the pypureclient being there is `count` is selected as that parameter is only supported in REST 2.2

Reply:
Ah I misunderstood. I'll approve the PR